packages/dd-trace/test/profiling/profilers/heap.spec.js

```diff
@@ -56,11 +63,68 @@ describe('profilers/native/heap', () => {
 
     profiler.start()
 
-    pprof.heap.profile.returns('profile')
-
     const profile = profiler.profile()
 
-    expect(profile).to.equal('profile')
+    expect(profile).to.equal(profileData)
+  })
+
+  it('should adjust heap profiler sample rate dynamically', () => {
+    let profileData
+    let profile
+
+    const profiler = new NativeHeapProfiler({
+      samplingInterval: 1024,
+      samplingThreshold: 0.5
+    })
+
+    profiler.start()
+
+    // Check sample rate adjusts up
+    pprof.heap.start.resetHistory()
+
+    profileData = {
+      sample: [{
+        value: [1, 1024 * 2 * 60 * 99]
+      }]
+    }
+
+    pprof.heap.profile.returns(profileData)
+    profile = profiler.profile()
+    expect(profile).to.equal(profileData)
```

Comment:
Minor: We repeat it from time to time, but since the function always returns the output of the pprof profiler, would it make sense to have it as a separate test case that just asserts that once?

Reply:
I'm changing the data each time though so this just verifies that the data I was expecting passed through it.